### ISSUE TITLE
Dashboard filters dropdown only list the first 100 values

### ISSUE THREAD

**Issue by lukeman** [metabase/metabase#15695]
**Describe the bug**
After the v0.39.0 upgrade dashboard filters for an ID field only seem to fetch the first 100 values (sorted alphabetically).

**Logs**
No real logs to speak of, but it appears the limitation is from the values endpoint (only 100 results returned). Example URL for a public dashboard: `/api/public/dashboard/616cbf50-d528-4847-8ba2-5dbe052180ba/params/f5acac2e/values`

**To Reproduce**
1. Have a dashboard with an ID field filter where possible values are >100
2. Search for values
3. Observe that only the first 100 alphabetical values are being searched against

**Expected behavior**
In prior versions all values were searchable, not just the first 100.

**Severity**
Especially when combined with the other dashboard filter bugs this is severe enough that I would consider it a blocker had I not already upgraded.

**Additional context**
Not sure if the DB connection matters, but this is with an ID field filter on a BigQuery table.

**Metabase Diagnostic Info**

```json
{
  "browser-info": {
    "language": "en-us",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.3 Safari/605.1.15",
    "vendor": "Apple Computer, Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.3+7-Debian-1bpo92",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "http://java.oracle.com/",
    "java.version": "11.0.3",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.3+7-Debian-1bpo92",
    "os.name": "Linux",
    "os.version": "4.19.0-16-amd64",
    "user.language": "en",
    "user.timezone": "America/Los_Angeles"
  },
  "metabase-info": {
    "databases": [
      "bigquery",
      "postgres",
      "presto",
      "mysql"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "9.5.7"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.18"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2021-04-19",
      "tag": "v0.39.0",
      "branch": "release-x.39.x",
      "hash": "5e40366"
    },
    "settings": {
      "report-timezone": "US/Pacific"
    }
  }
}
```

**Comment by flamber** [comment]
Not sure why this was limited to 100 in frontend, but we're requesting up to 1000 rows in the query, while the field scan uses 5000.
Related to #13832 and #7603

**Comment by lukeman** [comment]
If anyone is interested in a workaround I've found that switching any affected fields to use a "Search box" instead of "A list of all values" generally allows folks to select from the full list and overall it seems close to the prior behavior (other than not firing off for the list of full values on focus?). It's worth a shot and might even be a more appropriate choice if your values lists have gotten as long as mine seem to have these days. 😄 

Admin > Data Model > _Your Dataset_ > _Your Field_ ⚙️ > Change filtering on this field to search

<img width="654" alt="Screen Shot 2021-05-06 at 10 17 26 PM" src="https://user-images.githubusercontent.com/13008/117390215-bdb7f880-aebb-11eb-828d-f0efe06cf9a8.png">

**Comment by ariya** [comment]
For additional info:

FieldValuesWidget component (the filter drop-down in UI) makes a call to DashboardApi.parameterValues endpoint (in the above example from flamber: `/api/dashboard/1/params/b08cf05d/values`) and this endpoint returns an array of 100 elements.

If the response is modified to serve more than 100 elements, than the front-end (FieldValuesWidget) will happily show that.

**Comment by camsaul** [comment]
Fixed by #16546
